QuizApp/QuizApp/Network/UserNetworkDataSource.swift

```diff
@@ -0,0 +1,31 @@
+class UserNetworkDataSource: UserNetworkDataSourceProtocol {
+
+    private var loginClient: LoginClientProtocol
+
+    private var loginResponseData: LoginResponseData!
+
+    init(loginClient: LoginClientProtocol) {
+        self.loginClient = loginClient
+    }
+
+    func logIn(username: String, password: String) async throws -> LoginResponseData {
+        return LoginResponseData(try await loginClient.logIn(username: username, password: password))
+    }
+
+}
+
+struct LoginResponseData {
+
+    let accessToken: String
+
+    init(_ loginResponse: LoginResponse) {
```

Comment:
By defining the LoginResponseData constructor like this, you have disabled the default constructor. Please remove this constructor or move to extension.